modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java

```diff
@@ -172,99 +154,43 @@ public void setEnvironment() throws Exception {
-        assertFalse(addCommentJsonObject.getBoolean("error"), "Error in Add Comment Response");*/
+        APIListDTO getAllAPIsResponse = apiStore.getAllPublishedAPIs();
+        assertTrue(APIMTestCaseUtils.isAPIAvailableInStore(apiIdentifier, getAllAPIsResponse));
+
+        HttpResponse getAPIResponse = apiPublisher.getAPI(apiID);
+        assertEquals(getAPIResponse.getResponseCode(), Response.Status.OK.getStatusCode(), "Error when " +
+                "retrieving API: " + apiName);
+        Gson gson = new Gson();
+        APIDTO api = gson.fromJson(getAPIResponse.getData(), APIDTO.class);
+        LinkedTreeMap endpointConfig = (LinkedTreeMap) api.getEndpointConfig();
+        List sandboxEndpoints = (ArrayList) endpointConfig.get("sandbox_endpoints");
+        List productionEndpoints = (ArrayList) endpointConfig.get("production_endpoints");
+
+        assertEquals(productionEndpoints.size(), 5, "Mismatch in the number of production endpoints");
+        assertEquals(sandboxEndpoints.size(), 5, "Mismatch in the number of sandbox endpoints");
+
+        assertEquals(endpointConfig.get("endpoint_type"), "load_balance", "Endpoint type mismatched");
+        assertEquals(endpointConfig.get("algoCombo"), "org.apache.synapse.endpoints.algorithms.RoundRobin",
+                "Endpoint selection algorithm mismatched");
+
+        org.wso2.am.integration.clients.store.api.v1.dto.APIDTO apiFromStore = apiStore.getAPI(apiID);
+        List<APIEndpointURLsDTO> endpointURLs = apiFromStore.getEndpointURLs();
+        APIURLsDTO urls = endpointURLs.get(0).getUrLs();
+
+        assertEquals(endpointURLs.get(0).getEnvironmentName(), Constants.GATEWAY_ENVIRONMENTS, "Mismatch in " +
+                "environment name");
+        assertEquals(endpointURLs.get(0).getEnvironmentType(), "hybrid", "Mismatch in environment type");
+        assertTrue(urls.getHttp() != null, "HTTP URL is not available");
+        assertTrue(urls.getHttps() != null, "HTTPS URL is not available");
+        assertEquals(urls.getHttp(), gatewayUrl + apiContext + "/" + apiVersion, "Error in HTTP " +
+                "endpoint URL");
+        assertEquals(urls.getHttps(), gatewayHTTPSURL + apiContext + "/" + apiVersion, "Error in HTTPS" +
+                "endpoint URL");
     }
 
     @AfterClass(alwaysRun = true)
     public void destroy() throws Exception {
-        apiProvider = publisherContext.getContextTenant().getContextUser().getUserName();
-        HttpResponse deleteApiResponse = apiPublisher.deleteAPI(apiName, apiVersion, apiProvider);
-        assertEquals(deleteApiResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
-                     "Error in Delete API Response");
-        JSONObject deleteApiJsonObject = new JSONObject(deleteApiResponse.getData());
-        assertFalse(deleteApiJsonObject.getBoolean("error"), "Error in Delete API");
-        super.cleanUp();
+        apiPublisher.deleteAPI(apiID);
+        super.cleanUpUsingRest();
```

Comment:
Lets use the previous cleanUp method as discussed.

Reply:
Fixed